File: src/ELibrary.Backend/LibraryApi/Domain/Entities/Author.cs
using System.ComponentModel.DataAnnotations;

namespace LibraryApi.Domain.Entities
{
    public class Author : BaseEntity
    {
        [Required]
        [MaxLength(256)]
        public string Name { get; set; } = default!;

        [Required]
        [MaxLength(256)]
        public string LastName { get; set; } = default!;

        public DateTime DateOfBirth { get; set; }

        public override void Copy(BaseEntity other)
        {
            if (other is Author otherAuthor)
            {
                this.Name = otherAuthor.Name;
                this.LastName = otherAuthor.LastName;
                this.DateOfBirth = otherAuthor.DateOfBirth;
            }
        }
    }
}

File: src/ELibrary.Frontend/src/app/modules/shared/domain/models/library/author.ts
export interface Author {
    id: number;
    name: string;
    lastName: string;
    dateOfBirth: Date;
}

export function getDefaultAuthor(): Author {
    return {
        id: 0,
        name: "",
        lastName: "",
        dateOfBirth: new Date()
    }
}